### ISSUE TITLE
[windows 7] show missing api-ms-win-shcore-scaling-l1-1-1.dll error.

### ISSUE THREAD

**Issue by liqping** [flutter/flutter#95376]
Is it not support Windows 7 in version 2.9.0-0.1.pre?

**Comment by maheshmnj** [comment]
Hi @liqping, flutter does support windows 7, As per the flutter docs https://docs.flutter.dev/get-started/install/windows

> Operating Systems: Windows 7 SP1 or later (64-bit), x86-64 based.
 
 Can you please share the complete error logs that you are seeing along with your output of `flutter doctor -v` and steps to reproduce the issue (if known).
 
 Thanks

**Comment by stuartmorgan** [comment]
Are you using any plugins, or does this happen just running the default app created by `flutter create` with no changes?

If the latter, what's the first version where this happens?

**Comment by FusoraTech** [comment]
Typical problem of Win7 not updated. Try this http://support.microsoft.com/kb/976932

**Comment by stuartmorgan** [comment]
> Typical problem of Win7 not updated.

That was not the cause the last time this error happened; previously it was due to unconditional use of an API that didn't exist in Windows 7.

**Comment by liqping** [comment]
1.use flutter create without any other plugins
2.flutter build windows
3.Run It in win7 sp1,show error dialog message. See error screen.
4.It happen only 2.9.0-0.1.pre;It works on fluttrer 2.8 and early. 
5.error screen
![QQ图片20211217095808](https://user-images.githubusercontent.com/62126718/146477302-9cb0ac5a-4f4b-465c-bf93-74f0446e9c26.png)

6.sample code
 'import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          // Column is also a layout widget. It takes a list of children and
          // arranges them vertically. By default, it sizes itself to fit its
          // children horizontally, and tries to be as tall as its parent.
          //
          // Invoke "debug painting" (press "p" in the console, choose the
          // "Toggle Debug Paint" action from the Flutter Inspector in Android
          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
          // to see the wireframe for each widget.
          //
          // Column has various properties to control how it sizes itself and
          // how it positions its children. Here we use mainAxisAlignment to
          // center the children vertically; the main axis here is the vertical
          // axis because Columns are vertical (the cross axis would be
          // horizontal).
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}'


flutter doctor -v
` Flutter (Channel beta, 2.9.0-0.1.pre, on Microsoft Windows [Version 10.0.19043.1348], locale zh-CN)
    • Flutter version 2.9.0-0.1.pre at D:\fluter
    • Upstream repository https://gitee.com/mirrors/Flutter.git
    • FLUTTER_GIT_URL = https://gitee.com/mirrors/Flutter.git
    • Framework revision 8f1f9c10f0 (2 days ago), 2021-12-14 13:41:48 -0800
    • Engine revision 234aca678a
    • Dart version 2.16.0 (build 2.16.0-80.1.beta)
    • DevTools version 2.9.1
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\liping\AppData\Local\Android\sdk
    • Platform android-31, build-tools 30.0.3
    • Java binary at: D:\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.10.3)
    • Visual Studio at D:\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.10.31424.327
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at D:\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1348]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 96.0.4664.110
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 96.0.1054.57

• No issues found!
`

**Comment by maheshmnj** [comment]
Thanks for the reproducible steps @liqping, Looks like this issue is specific to windows 7, I am unable to reproduce this on Windows 10. on `master, 2.9.0-1.0.pre.95`.

<details>
<summary>flutter doctor -v</summary>

```
[√] Flutter (Channel master, 2.9.0-1.0.pre.95, on Microsoft Windows [Version 10.0.19042.1348],
    locale en-US)
    • Flutter version 2.9.0-1.0.pre.95 at C:\flutter_sdk\master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 935d1478aa (2 days ago), 2021-12-14 23:04:05 -0800
    • Engine revision 57f7719e6e
    • Dart version 2.16.0 (build 2.16.0-109.0.dev)
    • DevTools version 2.9.1

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)    
    • Android SDK at C:\Users\mahesh\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)        
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.4)      
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31727.386
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.61.2)
    • VS Code at C:\Users\mahesh\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.29.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.1348]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 96.0.4664.110
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 96.0.1054.53

• No issues found!
```
</details>

**Comment by stuartmorgan** [comment]
Ah, this is actually the same API as last time. @cbracken this is due to https://github.com/flutter/engine/pull/30013

See dpi_utils_win32 for the Windows-7-safe version. (You'll need to remove the direct linking against shcore as well.)

**Comment by liqping** [comment]
@stuartmorgan Need to modify flutter engine and then recompile?

**Comment by sgsm74** [comment]
Hi
same issue on windows 7 and 2.9.0-0.1.pre master, but in release mode...
any suggestion?

**Comment by cbracken** [comment]
@stuartmorgan Thanks for diagnosing. Will get a patch together.

**Comment by cbracken** [comment]
Fixed in flutter/engine#30699 on master branch.

**Comment by cbracken** [comment]
Added `CP: 2.10` label. We will need https://github.com/flutter/engine/pull/30699 cherry-picked into the 2.10 release.
